java/src/test/java/com/saucelabs/simplesauce/SauceSessionTest.java

```diff
@@ -43,15 +46,31 @@ public void sauceSession_defaultSauceOptions_returnsChromeBrowser() {
 
     @Test
     public void startSession_defaultConfig_usWestDataCenter() {
-        String expectedDataCenterUrl = DataCenter.US_WEST.getEndpoint();
-        assertEquals(expectedDataCenterUrl, sauce.getSauceDataCenter());
+        String expectedDataCenterEndpoint = DataCenter.US_WEST.getEndpoint();
+        assertEquals(expectedDataCenterEndpoint, sauce.getSauceDataCenter());
+    }
+
+    @Test
+    public void defaultSauceURL() throws MalformedURLException {
+        String dataCenterEndpoint = DataCenter.US_WEST.getEndpoint();
+        String user = System.getenv("SAUCE_USERNAME");
+        String key = System.getenv("SAUCE_ACCESS_KEY");
+        URL expetedSauceUrl = new URL("https://" + user + ":" + key + "@" + dataCenterEndpoint + ":443/wd/hub");
+        assertEquals(expetedSauceUrl, sauce.getSauceUrl());
+    }
+
+    @Test
+    public void setsSauceURLDirectly() throws MalformedURLException {
+        sauce.setSauceUrl(new URL("http://example.com"));
```

Comment:
I have some confusion here. Why would someone set a "Sauce URL", is that really what a user cares about? Or does the user care about setting the data center?
`sauce.setSauceDataCenter(DataCenter.UsWest)`
Regardless of the approach, I'd want an enum holding those values and avoid potential errors from strings.

Reply:
This is an easy answer. Schwab needed to set the URL. They had a relay and it needed to be overridden. :)

Reply:
Do you agree that's more of an exception than the rule? Would it make sense to have 2 methods then? `setSauceUrl` and `setSauceDataCenter`? The first for Shwab, the latter for 99% of other users?

Reply:
hmm, I'm not sure what your concern is here, this is what we have right now. Schwab can `setSauceUrl` which we don't recommend. Everyone else sets username, access key & data center.

Reply:
So there will the be capability to `setDataCenter()` somewhere? Guessing on `SauceOptions` ? So that someone with a normal use case doesn't have to pass a URL string?

Reply:
Oh, good call. Let's add a @Setter to that field, I thought it was already there. Which... brings me to another point in that the field should be the enum for the purposes of the setter... grrr, let me add another commit.